### ISSUE TITLE
Missing Region Server

### ISSUE THREAD

**Issue by shahamit** [kakao/hbase-region-inspector#2]
I have installed the latest version of hbase-region-inspector (hbase-region-inspector-0.3.3) and pointed it to the zookeeper client port. I have the web server running. I am facing few issues

1. I am not able to see one of the region servers that is part of the cluster. I can see that server on the hbase web ui but not on hbase-region-inspector. I noticed that this region server does not have any regions on it yet. Could that be the reason why it is not getting shown on the inspector? Due to this I am not able to move regions across servers.

2. The region count for a table is shown as zero when there is one region for that table created on the region server (verified from the hbase web ui). Could this be because that table is too small ? 

3. Clicking on the region server opens up a web page to <region-server>:16030 which I assume is the hbase region server http server port. I have deployed hbase on cloudera which didn't start the http server. Do I need to start it explicitly?

Thanks,
Amit

**Comment by junegunn** [comment]
> I noticed that this region server does not have any regions on it yet. Could that be the reason why it is not getting shown on the inspector?

Correct. That's currently how it works, but yeah I guess it makes more sense to just display the servers unconditionally. I'll change that shortly.

> The region count for a table is shown as zero when there is one region for that table created on the region server.

No, region count shouldn't be affected by the size of the regions and I can't reproduce the problem locally, so, can you post a screenshot of the page? Number of storefiles can be zero though if the table is new and was never flushed.

> Do I need to start it explicitly?

Web page for each regionserver starts automatically unless you explicitly set `hbase.regionserver.info.port` to -1 (which I doubt you did). 16030 is the default port since HBase 1.0, but Cloudera still uses old 60030 for backward compatibility. Try changing the port or the URL to 60030 and see if it opens. The reason hbase-region-inspector uses the default port even when the actual port is different is because there is no API to retrieve the actual port being used. One thing you can do is to create a properties file such as follows and use it as the first argument to hbase-region-inspector.

```
hbase.zookeeper.quorum = localhost
hbase.regionserver.info.port = 60030
```

**Comment by shahamit** [comment]
Appreciate your quick response. 

About the second point - region count shown as zero, kindly find the screenshot here - http://i.imgur.com/KbvVD1r.png

Also here's a screenshot from the hbase web ui - http://i.imgur.com/ixCYXpU.png. Note that even though the UI doesn't show the start and end key of region I can still query the table through PHOENIX sqlline client. The table has 1 row. 

Let me know if you find the root cause.

**Comment by junegunn** [comment]
Ah, you're looking at "Data size" tab, so the zeros are not the number of regions but the size of the regions in megabytes. And apparently the size of your table is less than 1MB so it's not showing. In this case, you might want to check "Storefiles" tab instead. Note that the table will still not be visible there if it was never flushed before (no storefile).

I'll release a new version with the changes we discussed here later in the day and let you know.

**Comment by junegunn** [comment]
Just released 0.3.4: https://github.com/kakao/hbase-region-inspector/releases/tag/v0.3.4
Try it and let me know if you have any issues. Thanks.
